backend/src/hatchling/bridge/app.py

```diff
@@ -110,12 +113,12 @@ def abort(self, message='', code=1, **kwargs):
 
         sys.exit(code)
 
-    def get_safe_application(self):
+    def get_safe_application(self) -> SafeApplication:
         return SafeApplication(self)
 
 
 class SafeApplication:
-    def __init__(self, app):
+    def __init__(self, app: InvokedApplication | Application) -> None:
```

Comment:
I believe a `typing.Protocol` would be more meaningful here, but the interfaces of `InvokedApplication` and `Application` are slightly different. @ofek, what do you think we should do? Would modifying the interfaces be appropriate or should this stay like this?

Reply:
I'm fine with changing the `InvokedApplication` to match. If so, please rename `message` to `text` to mirror https://github.com/pypa/hatch/blob/master/src/hatch/cli/terminal.py

Reply:
Actually trying to make this change I realise it wouldn't offer and it would make the interfaces less clean. The union is probably clearer in what it means, so I am leaving it as is unless you have other suggestions!